File: python/benchmark/benchlog.py
import sys, os, re

pattern = re.compile(r'^(\w[-\w]*)\s+([.\d]+)\s+([.\d]+)\s+([.\d]+)\s+([.\d]+)')

names = []
table = {}

for line in sys.stdin.readlines():
    m = pattern.match(line)
    if m:
        name, utime, stime, total, real = m.groups()
        #print name, utime, stime, total, real
        list = table.get(name)
        if not list:
            table[name] = list = []
            names.append(name)
        list.append((float(utime), float(stime), float(total), float(real), ))

print "                           utime      stime      total       real"
for name in names:
    tuples = table[name]
    utime = stime = total = real = 0.0
    for i, tuple in enumerate(tuples):
        utime += tuple[0]
        stime += tuple[1]
        total += tuple[2]
        real  += tuple[3]
    #print "%-21s %10.5f %10.5f %10.5f %10.5f" % (name, utime, stime, total, real)
    n = i + 1
    print "%-21s %7.2f000 %7.2f000 %7.2f000 %10.5f" % (name, utime/n, stime/n, total/n, real/n)

File: php/benchmark/benchlog.php
<?php
$data = array();
while (($line = fgets(STDIN)) !== false) {
    if (preg_match('/([-\w]+)\s+([\d.]+)\s+([\d.]+)\s+([\d.]+)\s+([\d.]+)/', $line, $m)) {
        $name = $m[1];
        if (! isset($data[$name])) $data[$name] = array();
        $data[$name][] = array(floatval($m[2]), floatval($m[3]), floatval($m[4]), floatval($m[5]));
    }
}

printf("%20s%10s %10s %10s %10s\n", ' ', 'user', 'sys', 'total', 'real');
foreach ($data as $name=>$tuples) {
    //var_export($item);
    $utime = $stime = $total = $real = 0.0;
    foreach ($tuples as $tuple) {
        $utime += $tuple[0];
        $stime += $tuple[1];
        $total += $tuple[2];
        $real  += $tuple[3];
    }
    //printf("%-20s%10.4f %10.4f %10.4f %10.4f\n", $name, $utime, $stime, $total, $real);
    //printf("%-20s%10.4f %10.4f %10.4f %10.4f\n", $name, $utime/10.0, $stime/10.0, $total/10.0, $real/10.0);
    printf("%-20s%8.2f00 %8.2f00 %8.2f00 %10.4f\n", $name, $utime/10.0, $stime/10.0, $total/10.0, $real/10.0);
}
?>